stanify/builders/stan_block_builder.py

```diff
@@ -0,0 +1,673 @@
+            self.code += ode_solver_code + ");\n"
+            self.code += "\n"
+
+            self.code += "// Assign target simulated to latent stock vectors\n"
+            for index, name in enumerate(integ_outcome_vector_names, 1):
+                self.code += f"{name}[:, r] = integrated_result[:, {index}];\n"
+            self.code.indent_level -= 1
+            self.code += "}\n"
+
+
+class StanModelBuilder:
+    def __init__(self, precision_context: "PrecisoinContext", stan_model_context: "StanModelContext"):
+        self.precision_context = precision_context
+        self.stan_model_context = stan_model_context
+
+    def build_block(self, hier_est_param_names):
+        code = IndentedString()
+        code += "model{\n"
+        code.indent_level += 1
+        if self.precision_context.R == 1:
+            for statement in self.stan_model_context.sample_statements:
+                if statement.distribution_type != statement.assignment_dist:
+                    code += f"{statement.lhs_expr} ~ {statement.distribution_type}({', '.join([str(arg) for arg in statement.distribution_args])});\n"
+        else:
+            for statement in self.stan_model_context.sample_statements:
+                param_name = statement.lhs_expr
+                if param_name in hier_est_param_names:
+                    dist_code = "rep_vector(" + f'{statement.distribution_args[0]}, R), ' + f"{', '.join(statement.distribution_args[1:])}"
+                    code += f"{param_name} ~ {statement.distribution_type}({dist_code});\n"
+
+                elif param_name in self.stan_model_context.obs_integ_outcome_vector_names:
+                    code += "for (r in 1:R)\n"
+                    code.indent_level += 1
+                    dist_code = f'{param_name}'[:-4] + "[:, r], " + f"{', '.join(statement.distribution_args[1:])}"
+                    code += f"{param_name}[:, r] ~ {statement.distribution_type}({dist_code});\n"
+                    code.indent_level -= 1
+
+                else:
+                    if statement.distribution_type != statement.assignment_dist:
+                        code += f"{param_name} ~ {statement.distribution_type}({', '.join([str(arg) for arg in statement.distribution_args])});\n"
+        code.indent_level -= 1
+        code += "}\n"
+        #TODO @Dashadower what is the diff btw classes that has its own code vs not (self.code VS return str(code))
+        return str(code)
+
+
+class Draws2DataStanGQBuilder():
+    def __init__(self, precision_context: "PrecisionContext", stan_model_context: "StanModelContext", vensim_model_context: "VensimModelContext"):
+        self.precision_context = precision_context
+        self.stan_model_context = stan_model_context
+        self.vensim_model_context = vensim_model_context
+        # TODO @Dashadower how to write message, if some target_simulated_vector_names is not in vensim_integ_outcome (inconsistency btw user-defined and vensim syntax)
```

Comment:
> how to write message, if some target_simulated_vector_names is not in vensim_integ_outcome (inconsistency btw user-defined and vensim syntax)

This should happen before block builders are called, say in `VensimModelContext`. The block builders will work under the assumption that inputs are sound.